ibis/expr/types/relations.py

```diff
@@ -247,6 +210,40 @@ def _bind_reduction_filter(self, where):
 
         return where.resolve(self)
 
+    def bind(self, *args, **kwargs):
+        # allow the first argument to be either a dictionary or a list of values
+        if len(args) == 1:
+            if isinstance(args[0], dict):
+                kwargs = {**args[0], **kwargs}
+                args = ()
+            else:
+                args = util.promote_list(args[0])
+
+        # bind positional arguments
+        values = []
+        for arg in args:
+            values.extend(bind(self, arg))
+
+        # bind keyword arguments where each entry can produce only one value
+        # which is then named with the given key
+        for key, arg in kwargs.items():
+            try:
+                (value,) = bind(self, arg)
+            except ValueError:
```

Comment:
This would gobble up unrelated errors: `t.bind(lambda t: int("foo"))`. Let's be a little slower here:
```
bindings = bind(self, arg)
if len(bindings) != 1:
    raise com.IbisInputError(
                    "Keyword arguments cannot produce more than one value"
                )
values.append(bindings[0].name(key))
```

Reply:
Yeah, we should avoid catching the error here. I'll fix it up.

Reply:
Done and test added.

Reply:
Nice catch! Thanks for the patch too!